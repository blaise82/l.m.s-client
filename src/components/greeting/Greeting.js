import React from 'react';
import {connect } from 'react-redux';
import {greetUser } from '../../redux/actions/greeting/greetingAction';
const greeting = (props) => {
    return (
        <div>
            <h4>{props.welcome}</h4>
        </div>
    );  
}

const mapStateToProps = ({greeting}) => {
    return {
        welcome: greeting.welcome
    }
}

const mapDispatchToProps = dispacth => {
    return {
        greetUser: () => dispacth(greetUser())
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(greeting);
