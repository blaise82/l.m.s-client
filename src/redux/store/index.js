import { createStore, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rootReducer from '../reducers';

const middlwares = applyMiddleware(logger, thunk);
const store =  createStore(rootReducer, composeWithDevTools(middlwares));

export default store;