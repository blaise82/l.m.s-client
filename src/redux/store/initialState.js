const state = {
            greeting: {
                welcome : 'Welcome to L.M.S - Your Library Managment System'
            }
          }
export default state;
