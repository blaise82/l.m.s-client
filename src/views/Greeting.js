import React from "react";
import { connect } from "react-redux";
import { Link, BrowserRouter as Router } from "react-router-dom";
import { greetUser } from "../redux/actions/greeting/greetingAction";
import Button from "../components/Button";
import classes from "../styles/Greeting.module.css";

const greeting = (props) => {
  return (
    <div>
      <h1 className={classes.Heading}>{props.welcome}</h1>
      <Link exact to="signin">
        <Button btnName="Sign In" />
      </Link>
      &nbsp;Or &nbsp;
      <Link exact to="signup">
        <Button btnName="Sign Up" />
      </Link>
    </div>
  );
};

const mapStateToProps = ({ greeting }) => {
  return {
    welcome: greeting.welcome,
  };
};

const mapDispatchToProps = (dispacth) => {
  return {
    greetUser: () => dispacth(greetUser()),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(greeting);
