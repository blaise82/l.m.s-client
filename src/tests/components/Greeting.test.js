import React from 'react'
import { Provider } from 'react-redux';
import {render} from '@testing-library/react'
import Greeting from '../../components/greeting/Greeting';
import store from '../../redux/store/index';
 
describe('Greeting component', () => {
    it('should render greeting component propely', () => {
        const { asFragment } = render( <Provider store={store}><Greeting /></Provider>)
        expect(asFragment(<Provider store={store}><Greeting /></Provider>)).toMatchSnapshot()
       })
    })
