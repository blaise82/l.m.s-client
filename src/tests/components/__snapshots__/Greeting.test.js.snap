// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeting component should render greeting component propely 1`] = `
<DocumentFragment>
  <div>
    <h4>
      Welcome to L.M.S - Your Library Managment System
    </h4>
  </div>
</DocumentFragment>
`;
